synthesis/fragment/fragment.go

```diff
@@ -34,6 +35,58 @@ func SetEfficiency(overhangs []string) float64 {
 	return efficiency
 }
 
+// NextOverhangs gets a list of possible next overhangs to use for an overhang
+// list, along with their efficiencies. This can be used for more optimal
+// fragmentation of sequences with potential degeneracy.
+func NextOverhangs(currentOverhangs []string) ([]string, []float64) {
+	currentOverhangMap := make(map[string]bool)
+	for _, overhang := range currentOverhangs {
+		currentOverhangMap[overhang] = true
+	}
+	bases := []rune{'A', 'T', 'G', 'C'}
+	var overhangsToTest []string
+	for _, base1 := range bases {
```

Comment:
okay I have to ask. What's with this super nested control flow? Can this be simplified?

Reply:
Not sure if simplified, but could probably be smaller by using a recursive function. I think it'd be more complex tho, I think this is reasonable cause you can clearly see what it is doing.

Reply:
Comment added.